resalloc_ibm_cloud/ibm_cloud_vm.py

```diff
@@ -106,14 +105,33 @@ def run_playbook(host, opts):
     subprocess.check_call(cmd, stdout=sys.stderr)
 
 
+def _get_zone_and_subnet_id(opts):
+    random_subnet = random.choice(opts.subnets_ids)
+    if ":" in random_subnet:
+        return tuple(random_subnet.split(":"))
+    return opts.zone, random_subnet
+
+
+def _get_private_ip_of_instance(instance_id, service):
+    for _ in range(300):
```

Comment:
really 300 attempts? :-)

Reply:
well we have only 1 VPN in IBM cloud and per chat with lmatousek, the VPN gets an update from time to time which means that VPN will be down for some time (I think it was approximately max 1 hour)

so I've added a waiting section here for 30 minutes. But it's not a whole hour - I don't know what would be better for us - if the VPN goes down, is it better to leave it as it is and let the builds fail or should we wait a whole hour to see if the VPN will start again?

Reply:
also, I am not sure what part will be affected if the VPN goes down (since it hasn't happened yet) - if the IP will be 0.0.0.0 since the machine is not up - unable to connect to VPN or it will magically go up but we won't be able to ssh to it (_wait_for_ssh will fail)